packages/smooth_app/lib/knowledge_panel/knowledge_panels/knowledge_panel_card.dart

```diff
@@ -48,4 +51,28 @@ class KnowledgePanelCard extends StatelessWidget {
       },
     );
   }
+
+  bool _isExpandedByUser(
+    final KnowledgePanel panel,
+    final KnowledgePanels allPanels,
+    BuildContext context,
+  ) {
+    final UserPreferences userPreferences = context.watch<UserPreferences>();
+    if (panel.titleElement != null &&
```

Comment:
A loop on `[EXPAND_PANEL_NUTRITION_TABLE_ID, EXPAND_PANEL_INGREDIENTS_ID]` would help avoiding duplicating the code, wouldn't it?